File: database/database-annotation-processor/src/test/java/ru/tinkoff/kora/database/common/annotation/processor/repository/error/InvalidParameterUsage.java
package ru.tinkoff.kora.database.common.annotation.processor.repository.error;

import ru.tinkoff.kora.database.common.annotation.Query;
import ru.tinkoff.kora.database.common.annotation.Repository;
import ru.tinkoff.kora.database.jdbc.JdbcRepository;

@Repository
public interface InvalidParameterUsage extends JdbcRepository {

    @Query("SELECT * FROM table WHERE field3 = :param1.someField")
    String wrongFieldUsedInTemplate(Dto param1, String param2);

    record Dto(String someField, String otherField) {}

}

File: database/database-symbol-processor/src/test/kotlin/ru/tinkoff/kora/database/symbol/processor/repository/error/InvalidParameterUsage.kt
package ru.tinkoff.kora.database.symbol.processor.repository.error

import ru.tinkoff.kora.database.common.annotation.Query
import ru.tinkoff.kora.database.common.annotation.Repository
import ru.tinkoff.kora.database.jdbc.JdbcRepository

@Repository
interface InvalidParameterUsage : JdbcRepository {
    @Query("SELECT * FROM table WHERE field3 = :param1.field3")
    fun wrongFieldUsedInTemplate(param1: Dto?, param2: String?): String?
    class Dto
}
